core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -321,7 +324,8 @@ public QuerySearchResult executeQueryPhase(QuerySearchRequest request) {
             operationListener.onPreQueryPhase(context);
             long time = System.nanoTime();
             queryPhase.execute(context);
-            if (context.queryResult().topDocs().scoreDocs.length == 0 && context.scrollContext() == null) {
+            if (context.queryResult().topDocs().scoreDocs.length == 0 && context.scrollContext() == null
```

Comment:
maybe we can factor out a method `boolean hasHits(QuerySearchResult result)` it's used in two places an a complex condition